internal/dcache/file_manager/utils.go

```diff
@@ -63,27 +67,67 @@ func isOffsetChunkStarting(offset int64, fileLayout *dcache.FileLayout) bool {
 	return (offset%fileLayout.ChunkSize == 0)
 }
 
+func getMVForChunk(chunk *StagedChunk, fileLayout *dcache.FileLayout) string {
+	numMvs := int64(len(fileLayout.MVList))
+	return fileLayout.MVList[chunk.Idx%numMvs]
+}
+
 // Does all file Init Process for creation of the file.
 func NewDcacheFile(fileName string) (*DcacheFile, error) {
 	fileMetadata := &dcache.FileMetadata{
 		Filename: fileName,
+		State:    dcache.Writing,
 	}
-	// todo : assign uuid for fileID
-	// todo : get chunkSize and Stripe Size from the config and assign.
-	// todo : Choose appropriate MV's from the online mv's returned by the clustermap
+	fileMetadata.FileID = gouuid.New().String()
+	common.Assert(common.IsValidUUID(fileMetadata.FileID))
+
+	chunkSize := clustermap.GetCacheConfig().ChunkSize
+	stripeSize := clustermap.GetCacheConfig().StripeSize
+
+	common.Assert(stripeSize%chunkSize == 0, fmt.Sprintf("Stripe Size %d is not divisibe by chunkSize %d", stripeSize, chunkSize))
+	numMVs := stripeSize / chunkSize
+
 	fileMetadata.FileLayout = dcache.FileLayout{
-		ChunkSize:  4 * 1024 * 1024,
-		StripeSize: 16 * 1024 * 1024,
-		MVList:     []string{"mv0", "mv1", "mv2"},
+		ChunkSize:  int64(chunkSize),
+		StripeSize: int64(stripeSize),
+		MVList:     make([]string, numMVs),
 	}
+
+	// Get active MV's from the clustermap
+	activeMvs := clustermap.GetActiveMVs()
```

Comment:
pls use MVs and not Mvs

Reply:
Instead of using GetActiveMVs(), let's add the following to clustermap and then use GetActiveMVNames() here

```
func GetActiveMVNames() []string {
        return clustermapImpl.getActiveMVNames()
}

func (c *ClusterMapImpl) getActiveMVNames() []string {
        common.Assert(c.localMap != nil)

        i := 0
        activeMVNames := make([]string, len(c.localMap.MVMap))
        for mvName, mv := range c.localMap.MVMap {
                if mv.State == dcache.StateOnline {
                        activeMVNames[i] = mvName
                        i++
                }
        }
        return activeMVNames[:i]
}
```